lib/kms/common.js

```diff
@@ -0,0 +1,165 @@
+import crypto from 'crypto';
+import { errors } from 'arsenal';
+
+/** Class exposing common createDataKey,
+createDecipher and createCipher functions. */
+export default class Common {
+
+    static _algorithm() {
+        return 'aes-256-ctr';
+    }
+
+    /* AES-256 Key */
+    static _keySize() {
+        return 32;
+    }
+
+    /* IV is 128bit for AES-256-CTR */
+    static _IVSize() {
+        return 16;
+    }
+
+    /* block size is 128bit for AES-256-CTR */
+    static _aesBlockSize() {
+        return 16;
+    }
+
+    /**
+    Creates data key to encrypt and decrypt the actual data (which data
+    key is ciphered and deciphered by the appliance). The encrypted data key
+    is stored in the object's metadata. We also use this function to create
+    the bucket key for the file and in memory implementations.
+    @return {buffer} - random key
+    */
+    static createDataKey() {
+        return new Buffer(crypto.randomBytes(this._keySize()));
+    }
+
+    /**
+     *
+     * @param {buffer} derivedIV - the stringified bucket
+     * @param {number} counter - quotient of the offset and blocksize
+     * @return {buffer} - the incremented IV
+     */
+    static _incrementIV(derivedIV, counter) {
+        const newIV = derivedIV;
+        const len = derivedIV.length;
+        let i = len - 1;
+        let ctr = counter;
+        while (ctr !== 0) {
+            const mod = (ctr + newIV[i]) % 256;
```

Comment:
I think the variable "mod' is not updated at each iteration, it's a const variable.
```es6
➜  ~ cat test.js 
var i = 0;
while (i < 10) {
	const a = i;
	++i;
	console.log(a);
}
➜  ~ node test.js 
0
0
0
0
0
0
0
0
0
0
```
```es6
➜  ~ cat test.js 
var i = 0;
while (i < 10) {
	var a = i;
	++i;
	console.log(a);
}
➜  ~ node test.js
0
1
2
3
4
5
6
7
8
9
```